### PR TITLE
Add !fps command

### PR DESCRIPTION
**Pull request by LiveFireExercise**
I wrote this really hastily and it probably doesn't work

### CODE REVIEW

File: MainModule/Client/UI/Default/FPS.lua [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+			Position = UDim2.new(0, 0, 0, 0);
+			--TextScaled = true;
+			--TextWrapped = true;
+		})
+		
+		gTable = window.gTable
+		window:Ready()
+		
+		repeat
+			label.Text = client.Remote.FPS().." Frames"
```
Comment (LolloDev5123) [inline MainModule/Client/UI/Default/FPS.lua:39]: This is inaccurate and spreading misinformation, as the FPS is not the amount of frames, but rather an approximation of the amount of frames displayed in a single second. Consider setting the text to only the number, and the Window Title to FPS or Frames per second, you don't need to clarify next to the number

**Review by LolloDev5123 — APPROVED** [review]

File: MainModule/Client/Core/Remote.lua [hunk 2]
```diff
@@ -505,6 +509,29 @@ return function(Vargs, GetEnv)
...
+			
+			local RenderStepped = service.RunService.RenderStepped
+			local Heartbeat = service.RunService.Heartbeat
+
+			local function GetFPS()
+				FPS_Counter+=1
+				if (os.clock() - startTime) >= X  then 
+					local fps = math.floor(FPS_Counter / (os.clock() - startTime))
+					return fps
+					FPS_Counter = 0
```
Comment (Expertcoderz) [inline MainModule/Client/Core/Remote.lua:528]: Please fix.

Comment (DimensionalFusion) [reply MainModule/Client/Core/Remote.lua:528]: no. you fix it if your code is so good

Comment (Expertcoderz) [reply MainModule/Client/Core/Remote.lua:528]: it's your PR, not mine; so you fix it if you are so convinced that your contribution is a good idea in the first place :p

Comment (P3tray) [reply MainModule/Client/Core/Remote.lua:528]: > nvm I'll fix it and get ez co-author :troll:

dude just close the PR and merge it yourself ezzzzzzz

File: MainModule/Client/UI/Default/FPS.lua [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+			--TextScaled = true;
+			--TextWrapped = true;
+		})
+
+		gTable = window.gTable
+		window:Ready()
+
+		repeat
+			label.Text = tostring(client.Remote.FPS())
+			wait(2)
```
Comment (P3tray) [inline MainModule/Client/UI/Default/FPS.lua:40]: dude you're really pausing execution in 2022? eww wtf.

File: MainModule/Client/UI/Default/FPS.lua [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	})
+
+	if window then
+		local label = window:Add("TextLabel", {
+			Text = "...";
+			BackgroundTransparency = 1;
+			TextSize = 20;
+			Size = UDim2.fromScale(1, 1);
+			Position = UDim2.new(0, 0, 0, 0);
+			--TextScaled = true;
```
Comment (P3tray) [inline MainModule/Client/UI/Default/FPS.lua:31]: bro remove the comments dude bro

File: MainModule/Client/UI/Default/FPS.lua [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+			--TextWrapped = true;
```
Comment (P3tray) [inline MainModule/Client/UI/Default/FPS.lua:32]: bro remove the comments dude bro

File: MainModule/Client/UI/Default/FPS.lua [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+	local measuring = true
+	local gTable
+
+	local window = client.UI.Make("Window", {
+		Name  = "FPS";
+		Title = "FPS";
+		Icon = client.MatIcons.Leaderboard;
+		Size  = {150, 70};
+		Position = UDim2.new(0, 10, 1, -80);
+		AllowMultiple = false;
```
Comment (P3tray) [inline MainModule/Client/UI/Default/FPS.lua:17]: 1984

File: MainModule/Client/UI/Default/FPS.lua [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+	end
+
+	local measuring = true
+	local gTable
+
+	local window = client.UI.Make("Window", {
+		Name  = "FPS";
+		Title = "FPS";
+		Icon = client.MatIcons.Leaderboard;
+		Size  = {150, 70};
```
Comment (P3tray) [inline MainModule/Client/UI/Default/FPS.lua:15]: shouldn't you be disallowing resize? specify a minsize perhaps?

File: MainModule/Client/UI/Default/FPS.lua [hunk 8]
```diff
@@ -0,0 +1,43 @@
+client, service = nil, nil
+
+return function(data, env)
+	if env then
+		setfenv(1, env)
```
Comment (P3tray) [inline MainModule/Client/UI/Default/FPS.lua:5]: dude this is cringe just directly use the passed env arguments and then define the services locally

File: MainModule/Client/Core/Remote.lua (hunk 2 above)
Comment (P3tray) [inline MainModule/Client/Core/Remote.lua:528]: even the pissing linter knows this codes fucked
image

**Review by P3tray — CHANGES_REQUESTED** [review]

**Comment by DimensionalFusion** [comment]
your mother

File: MainModule/Server/Commands/Players.lua [hunk 9]
```diff
@@ -309,6 +309,17 @@ return function(Vargs, env)
 				Remote.MakeGui(plr, "Ping")
 			end
 		};
+																		
+		FPS = {
+			Prefix = Settings.PlayerPrefix;
+			Commands = {"fps", "framespersecond"};
```
[the remark this answers is not in the document]
Comment (Expertcoderz) [reply MainModule/Server/Commands/Players.lua:315]: Lmao, same for aliases like :permanentadmin when :padmin exists.

[the remark this answers is not in the document]
Comment (DimensionalFusion) [reply MainModule/Server/Commands/Players.lua:315]: real

File: MainModule/Server/Commands/Players.lua [hunk 10]
```diff
@@ -309,6 +309,17 @@ return function(Vargs, env)
...
+			Args = {};
+			Description = "Shows you your current framerate";
```
[the remark this answers is not in the document]
Comment (LolloDev5123) [reply MainModule/Server/Commands/Players.lua:317]: "Shows you your number of frames generated for the monitor to display per second"

D*on't* you mean

"Shows the number of frames displayed per second"

🤓🤓🤓

**Comment by LolloDev5123** [comment]
> your mother

unoriginal reply ngl
Fix the parsing error by resetting FPS_Counter before returning the FPS, and possibly add a min size or remove resizing altogether

**Comment by P3tray** [comment]
when if you fix??

**Comment by P3tray** [comment]
@Expertcoderz add the wontfix label ¯\_(ツ)_/¯

@DimensionalFusion fix

**Comment by DimensionalFusion** [comment]
No, I don't think I will.

File: MainModule/Client/Core/Remote.lua [hunk 11]
```diff
@@ -505,6 +509,29 @@ return function(Vargs, GetEnv)
...
+					startTime = os.clock()
+				end
+			end
+			local FPS = Heartbeat:Connect(GetFPS)
+			return FPS
+		end;
```
Comment (ccuser44) [inline MainModule/Client/Core/Remote.lua:534]: What this basically does is that this only checks the checks the HeartBeat FPS which is the same as the physics FPS.
In which case you should replace this function with `workspace:GetRealPhysicsFPS()`.

File: MainModule/Client/UI/Default/FPS.lua [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+			Position = UDim2.new(0, 0, 0, 0);
+			--TextScaled = true;
+			--TextWrapped = true;
+		})
+
+		gTable = window.gTable
+		window:Ready()
+
+		repeat
+			label.Text = tostring(client.Remote.FPS())
```
Comment (ccuser44) [inline MainModule/Client/UI/Default/FPS.lua:39]: This only checks the heartbeat FPS which should just be replaced `workspace.GetRealPhysicsFPS()`. Unless you replace the function to use `RenderStepped`

**Review by ccuser44 — CHANGES_REQUESTED** [review]
I'm not sure if we need an :fps command. `Ctrl + Shift + F5` can easily achieve the same result with more information (it shows network ping send data, Render FPS and physics FPS).

**Also this only shows the physics/heartbeat FPS! NOT THE RENDER FPS WHICH IS HOW MANY TIMES IT RENDERS ON YOUR SCREEN!**. If you wan't it to show the Render FPS then you should use `.RenderStepped` otherwise you should just use `workspace:GetRealPhysicsFPS()`

**Comment by P3tray** [comment]
Finally, closure at long last.

**Pull request closed without merging**